graphs-lab/src/main/kotlin/graphs_lab/algs/TarjanBridgeFinding.kt

```diff
@@ -0,0 +1,87 @@
+package graphs_lab.algs
+
+import graphs_lab.algs.utils.checkAndGetFirst
+import graphs_lab.algs.utils.checkAndGetSecond
+import graphs_lab.algs.utils.removeAndReturn
+import graphs_lab.core.edges.Edge
+import graphs_lab.core.graphs.Graph
+
+/**
+ * The Tarjan's bridge-finding algorithm.
+ *
+ * Finding the bridges in an undirected graph.
+ * A bridge is an edge of a graph whose deletion increases the graph's number of connected components.
+ *
+ * @references https://en.wikipedia.org/wiki/Bridge_(graph_theory)
+ *
+ * @param I the type of the vertex identifiers
+ * @param E the type of the edges in the graph
+ * @property graph the input graph
+ */
+class TarjanBridgeFinding<I, E : Edge<I>>(
+	val graph: Graph<I, E>
+) {
+	private val setBridges = mutableSetOf<Edge<I>>()
+
+	/**
+	 * [tableHeights] the map that associates the identifier of a vertex with a pair that consist of
+	 * the minimal height (in an imaginary tree) among the vertices connected by an inverse edge (in an imaginary tree)
+	 * to the vertices from the subtree with the associated vertex in root or the height (in an imaginary tree)
+	 * of the associated vertex and the height (in an imaginary tree) of the associated vertex.
+	 */
+	private val tableHeights = mutableMapOf<I, Pair<Int?, Int?>>()
+	private val listUnvisitedVertices = mutableListOf<I>()
+
+	init {
+		require(!graph.isDirected) { "Do not support directed graph." }
+	}
+
+	/**
+	 * @return [setBridges]
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getBridges(): Set<Edge<I>> {
```

Comment:
Пара замечаний
1. Где описание функции, даже простейшее: `find all bridges and returns it`;
2. Согласно документации этой функции утверждается что производится возврат `private` переменной которая не документирована. Считаю что пользователю это излишне, а так же сама функция несовсем соответствует такой документации.

В связи с этим предлагаю:
```suggestion
	 /**
	 * Find all bridges by running dfs.
	 * 
	 * @return set of all bridges in graph
	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
	 */
	fun getBridges(): Set<Edge<I>> {
```